packages/faustwp-core/src/components/Toolbar/Toolbar.tsx

```diff
@@ -0,0 +1,149 @@
+import React, { useEffect, useMemo, useState } from 'react';
+import { Faust, Edit, GraphiQL } from './nodes/index.js';
+import { hooks } from '../../wpHooks/index.js';
+import { ToolbarItem } from './ToolbarItem.js';
+import { SeedNode } from '../../queries/seedQuery.js';
+
+/**
+ * The available menu locations that nodes can be added to.
+ */
+export type MenuLocation = 'primary' | 'secondary';
+
+/**
+ * A single Faust Toolbar node.
+ */
+export type FaustToolbarNode = {
+  /**
+   * The key identifier for this Toolbar Node.
+   * Used to create each Toolbar Node's id. `#wp-admin-bar-{key}`
+   */
+  key: string;
+
+  /**
+   * The available menu locations that nodes can be added to.
+   */
+  location: MenuLocation;
+
+  /**
+   * The JSX component to load.
+   * This will ultimately be rendered inside of an `<li>`.
+   */
+  component: React.ReactElement;
+};
+
+/**
+ * An array of toolbar nodes.
+ */
+export type FaustToolbarNodes = FaustToolbarNode[];
+
+/**
+ * Toolbar context.
+ */
+export type FaustToolbarContext = {
+  seedNode: SeedNode;
+};
+
+/**
+ * Toolbar props.
+ */
+export type ToolbarProps = {
+  client: React.ReactNode;
+  seedNode: SeedNode;
+};
+
+/**
+ * Renders a Toolbar that is based on WordPress' own toolbar.
+ */
+export function Toolbar({ client, seedNode }: ToolbarProps) {
+  /**
+   * Define Toolbar Nodes that should be included by default.
+   */
+  const coreToolbarNodes: FaustToolbarNodes = useMemo(() => {
+    return [
+      {
+        key: 'faust',
+        location: 'primary',
+        component: <Faust />,
+      },
+      {
+        key: 'edit',
+        location: 'primary',
+        component: <Edit seedNode={seedNode} />,
+      },
+      {
+        key: 'graphiql',
+        location: 'primary',
+        component: <GraphiQL />,
+      },
+    ];
+  }, [seedNode]);
+
+  const [toolbarNodes, setToolbarNodes] = useState(coreToolbarNodes);
+
+  /**
+   * Handle Toolbar nodes.
+   */
+  useEffect(() => {
+    const filteredNodes = hooks.applyFilters('toolbarNodes', coreToolbarNodes, {
+      client,
+      seedNode,
+    }) as FaustToolbarNodes;
+
+    const uniqueKeys = new Set(filteredNodes.map((nodes) => nodes.key));
+
+    if (uniqueKeys.size < filteredNodes.length) {
+      throw new Error('Toolbar Nodes must have unique keys.');
```

Comment:
Nice! We should also surface the `uniqueKeys` so users can easily determine what keys are not unique

Reply:
@blakewilson Appreciate the feedback! 

This guard clause prevents the app from running if a user attempts to add a Toolbar node with an existing key (via @wordpress/hooks). Surfacing `uniqueKeys` feels unnecessary here but I could be misunderstanding a scenario.

Reply:
I was just thinking in the case, for example, a user downloads a toolbar node plugin from npm and installs it, they may not necessarily know what unique IDs are colliding since they didn't built the toolbar plugin themselves. Knowing which IDs collided will allow them to notify the package maintainer of the issue.